src/visual.ts

```diff
@@ -392,19 +392,28 @@ module powerbi.extensibility.visual {
 
             excludedSet.forEach((item: PrimitiveValue) => {
                 if (typeof item === "string" || typeof item === "number") {
-                    let splittedExclude: string[] = item.toString()
-                        .replace(punctuationRegExp, " ")
-                        .split(whiteSpaceRegExp);
+                    let splittedExclude: string[] = [];
+
+                    // Filters should keep the same rules that target words
+                    if (!settings.general.isPunctuationsCharacters) {
+                        splittedExclude = item.toString().replace(punctuationRegExp, " ").split(whiteSpaceRegExp);
+                    } else {
+                        splittedExclude = item.toString().split(whiteSpaceRegExp);
+                    }
 
                     splittedExcludes = splittedExcludes.concat(splittedExclude);
                 }
             });
 
             words.forEach((item: WordCloudText) => {
                 if (typeof item.text === "string") {
-                    let splittedWords: string[] = item.text
-                        .replace(punctuationRegExp, " ")
-                        .split(whiteSpaceRegExp);
+                    let splittedWords: string[] = [];
+                    
+                    if (!settings.general.isPunctuationsCharacters) {
+                        splittedWords = item.text.replace(punctuationRegExp, " ").split(whiteSpaceRegExp);
+                    } else {
+                        splittedWords = item.text.split(whiteSpaceRegExp);
+                    }
```

Comment:
This code below is copied at a few places of visual.ts. It'd be better to move this code into a new function.

```TypeScript
let splittedWords: string[] = [];

if (!settings.general.isPunctuationsCharacters) {
    splittedWords = item.text.replace(punctuationRegExp, " ").split(whiteSpaceRegExp);
} else {
    splittedWords = item.text.split(whiteSpaceRegExp);
}
```